enterprise/cmd/frontend/internal/executorqueue/metrics/gcp.go

```diff
@@ -0,0 +1,91 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	monitoring "cloud.google.com/go/monitoring/apiv3/v2"
+	"github.com/cockroachdb/errors"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/inconshreveable/log15"
+	"google.golang.org/api/option"
+	metricpb "google.golang.org/genproto/googleapis/api/metric"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+
+	"github.com/sourcegraph/sourcegraph/enterprise/cmd/frontend/internal/executorqueue/handler"
+	"github.com/sourcegraph/sourcegraph/internal/workerutil/dbworker/store"
+)
+
+func initGCPMetrics(config *GCPConfig, queueOptions map[string]handler.QueueOptions) error {
+	if config.ProjectID == "" {
+		return nil
+	}
+
+	metricClient, err := monitoring.NewMetricClient(context.Background(), gcsClientOptions(config)...)
+	if err != nil {
+		return err
+	}
+
+	for queueName, options := range queueOptions {
+		initGCPMetric(config, metricClient, queueName, options.Store)
+	}
+
+	return nil
+}
+
+func initGCPMetric(config *GCPConfig, metricClient *monitoring.MetricClient, queueName string, store store.Store) {
+	go func() {
+		for {
+			if err := sendGCPMetric(config, metricClient, queueName, store); err != nil {
+				log15.Error("Failed to send executor queue size metric to GCP", "queue", queueName, "error", err)
+			}
+
+			time.Sleep(time.Second * 5)
+		}
+	}()
+}
+
+func sendGCPMetric(config *GCPConfig, metricClient *monitoring.MetricClient, queueName string, store store.Store) error {
+	count, err := store.QueuedCount(context.Background(), nil)
+	if err != nil {
+		return errors.Wrap(err, "dbworkerstore.QueuedCount")
+	}
+
+	if err := metricClient.CreateTimeSeries(context.Background(), makeGCPMetricRequest(config, queueName, count)); err != nil {
+		return errors.Wrap(err, "metricClient.CreateTimeSeries")
+	}
+
+	return nil
+}
+
+const (
+	gcpMetricKind = metricpb.MetricDescriptor_GAUGE
+	gcpMetricType = "custom.googleapis.com/executors/queue/size"
+)
+
+func makeGCPMetricRequest(config *GCPConfig, queueName string, count int) *monitoringpb.CreateTimeSeriesRequest {
+	pbMetric := &metricpb.Metric{Type: gcpMetricType, Labels: map[string]string{"queueName": queueName}}
+	now := &timestamp.Timestamp{Seconds: time.Now().UTC().Unix()}
```

Comment:
I _think_ this does it automatically

```suggestion
	now := &timestamp.Timestamp{Seconds: timeutil.Now().Unix()}
```